File: ProjetoEventX/DTOs/Auth/RegisterRequestDto.cs
using System.ComponentModel.DataAnnotations;

namespace ProjetoEventX.DTOs.Auth
{
    public class RegisterRequestDto
    {
        [Required]
        [StringLength(120, MinimumLength = 2)]
        public string NomeCompleto { get; set; } = string.Empty;

        [Required]
        [EmailAddress]
        [StringLength(256)]
        public string Email { get; set; } = string.Empty;

        [Required]
        [StringLength(120, MinimumLength = 6)]
        public string Password { get; set; } = string.Empty;

        [Required]
        [StringLength(30)]
        public string TipoUsuario { get; set; } = "Organizador";

        [StringLength(120)]
        public string? UserName { get; set; }

        [StringLength(14)]
        public string? Cpf { get; set; }

        [StringLength(18)]
        public string? Cnpj { get; set; }

        [StringLength(255)]
        public string? Endereco { get; set; }

        [StringLength(30)]
        public string? Telefone { get; set; }

        [StringLength(100)]
        public string? Cidade { get; set; }

        [StringLength(2)]
        public string? Uf { get; set; }

        [StringLength(255)]
        public string? TipoServico { get; set; }
    }
}

File: ProjetoEventXFlutter/lib/features/auth/data/models/register_request_dto.dart
class RegisterRequestDto {
  const RegisterRequestDto({
    required this.nomeCompleto,
    required this.email,
    required this.password,
    required this.tipoUsuario,
    this.userName,
    this.cpf,
    this.cnpj,
    this.endereco,
    this.telefone,
    this.cidade,
    this.uf,
    this.tipoServico,
  });

  final String nomeCompleto;
  final String email;
  final String password;
  final String tipoUsuario;
  final String? userName;
  final String? cpf;
  final String? cnpj;
  final String? endereco;
  final String? telefone;
  final String? cidade;
  final String? uf;
  final String? tipoServico;

  Map<String, dynamic> toJson() {
    final mappedUserType = _mapUserType(tipoUsuario);
    return {
      'fullName': nomeCompleto,
      'email': email,
      'password': password,
      'userType': mappedUserType,
    }..removeWhere((key, value) =>
        value == null || (value is String && value.trim().isEmpty));
  }

  static String _mapUserType(String value) {
    final normalized = value.trim().toLowerCase();
    switch (normalized) {
      case 'organizador':
      case 'organizer':
        return 'Organizer';
      case 'fornecedor':
      case 'supplier':
        return 'Supplier';
      case 'convidado':
      case 'guest':
      default:
        return 'Guest';
    }
  }
}
